### ISSUE TITLE
RPM build for OL7/Fedora 23 fail for docker-1.11.0 due to dwz: dwz.c:1984: checksum_die

### ISSUE THREAD

**Issue by thtanaka** [docker/docker#22051]
while building for OL7 RPM, we fail during the debuginfo phase:
``+ /usr/lib/rpm/find-debuginfo.sh --strict-build-id -m --run-dwz --dwz-low-mem-die-limit 10000000 --dwz-max-die-limit 110000000 /root/rpmbuild/BUILD/docker-engine
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160412.080258.gitd070eeb.el7.x86_64/usr/bin/docker
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160412.080258.gitd070eeb.el7.x86_64/usr/bin/docker-containerd
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160412.080258.gitd070eeb.el7.x86_64/usr/bin/docker-containerd-shim
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160412.080258.gitd070eeb.el7.x86_64/usr/bin/docker-containerd-ctr
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160412.080258.gitd070eeb.el7.x86_64/usr/bin/docker-runc
dwz: dwz.c:1984: checksum_die: Assertion `((!(multifile_mode & MULTIFILE_MODE_OP) && !(multifile_mode & MULTIFILE_MODE_RD) && !(multifile_mode & MULTIFILE_MODE_FI)) || cu != die_cu (ref)) && (!(multifile_mode & MULTIFILE_MODE_OP) || cu->cu_chunk == die_cu (ref)->cu_chunk)' failed.
/usr/lib/rpm/find-debuginfo.sh: line 361:  3453 Aborted                 (core dumped) dwz $dwz_opts $dwz_files
/usr/lib/rpm/sepdebugcrcfix: Updated 5 CRC32s, 0 CRC32s did match.
``
This is seems to be also happening on Fedora 23 RPM build.
According to this article: https://fedoraproject.org/wiki/PackagingDrafts/Go 
This could be due to the DWZ is unable to handle multiple golang binaries, as per the following bug
https://bugzilla.redhat.com/show_bug.cgi?id=995136#c12

By adding the following to hack/make/.build-rpm/docker-engine.spec, seems to make it work
``%global _dwz_low_mem_die_limit 0``

I opened this issue just to see if there is anyone here is expert on the DWZ and could offer a more viable solution. Thanks !

**Comment by thaJeztah** [comment]
Are you building from the v1.11 tag? What kind of setup are you building on? Our packages are built using that source, so there must be something different in your setup

**Comment by thtanaka** [comment]
Yes, from v1.11.0

The following is the head commit log:
commit 4dc5990d7565a4a15d641bc6a0bc50a02cfcf302
Merge: 365d80b 2535db8
Author: Tibor Vass <tiborvass@users.noreply.github.com>
Date:   Wed Apr 13 14:03:42 2016 -0400

    Merge pull request #22006 from tiborvass/fix-changelog-1.11

    Fix some CHANGELOG entries

Well, this is the docker info output from my setup:
```
# docker info
Containers: 6
 Running: 0
 Paused: 0
 Stopped: 6
Images: 142
Server Version: 1.10.2
Storage Driver: btrfs
 Build Version: Btrfs v3.19.1
 Library Version: 101
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.1.12-32.el7uek.x86_64
Operating System: Oracle Linux Server 7.2
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 47.16 GiB
```

**Comment by thtanaka** [comment]
Centos 7 didn't hit this issue because it is already setting the ``dwz-low-mem-die-limit`` to 0, the following is the build output of centos 7:
```
+ /usr/lib/rpm/find-debuginfo.sh --strict-build-id -m --run-dwz --dwz-low-mem-die-limit 0 --dwz-max-die-limit 110000000 /root/rpmbuild/BUILD/docker-engine
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160414.080249.gitef2e04f.el7.centos.x86_64/usr/bin/docker
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160414.080249.gitef2e04f.el7.centos.x86_64/usr/bin/docker-containerd
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160414.080249.gitef2e04f.el7.centos.x86_64/usr/bin/docker-containerd-shim
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160414.080249.gitef2e04f.el7.centos.x86_64/usr/bin/docker-containerd-ctr
extracting debug info from /root/rpmbuild/BUILDROOT/docker-engine-1.11.0-0.0.20160414.080249.gitef2e04f.el7.centos.x86_64/usr/bin/docker-runc
```
